src/main/webapp/Homepage.js

```diff
@@ -0,0 +1,9 @@
+submitAllForms = function(){
```

Comment:
The homepage does not have any forms so I am wondering why we need this JS.
I don't see any of the id's either in the html. Did you mean that for a different component?